src/Task/Ecs.php

```diff
@@ -52,16 +54,29 @@ public function run(ContextInterface $context): TaskResultInterface
     {
         $config = $this->getConfiguration();
 
+        /** @var array $whitelistPatterns */
+        $whitelistPatterns = $config['whitelist_patterns'];
+
         $files = $context->getFiles()->extensions($config['triggered_by']);
+        if (\count($whitelistPatterns)) {
+            $files = $files->paths($whitelistPatterns);
+        }
+
         if (0 === \count($files)) {
             return TaskResult::createSkipped($this, $context);
         }
 
         $arguments = $this->processBuilder->createArgumentsForCommand('ecs');
         $arguments->add('check');
 
-        foreach ($config['whitelist_patterns'] as $whitelistPattern) {
-            $arguments->add($whitelistPattern);
+        if ($config['files_on_pre_commit'] && $context instanceof GitPreCommitContext) {
+            $arguments->addFiles($files);
+        }
+
+        if (!$config['files_on_pre_commit'] && $context instanceof RunContext) {
```

Comment:
This statement doesn't seem right: in run context it should always use the whitelist instead of the files?